### PR TITLE
Addy gps

### PR DESCRIPTION
**Pull request by addy10**
Pull Request Overview

This pull request adds the gps sketch to the main example

This pull request was tested by using Catena board 4610 and the helium staging console.

TODO or Help Wanted

### CODE REVIEW

**Comment by lthiery** [comment]
Just starting to look into the details, but certainly create a fresh sketch for the GPS app. Leave the simple app as a sample @addy10

File: longfi-us915/longfi-us915.ino [hunk 1]
```diff
@@ -35,6 +35,15 @@
 #include <arduino_lmic.h>
 #include <hal/hal.h>
 #include <SPI.h>
+#include <Adafruit_GPS.h>
+
+#define GPSSerial Serial1
+
+#define CFG_sx1276_radio 1
```
Comment (lthiery) [inline longfi-us915/longfi-us915.ino]: is this necessary? we probably don't want to do this

File: longfi-us915/longfi-us915.ino [hunk 2]
```diff
@@ -232,11 +241,46 @@ void do_send(osjob_t* j){
...
+        uint32_t data;
+
+        if (GPS.newNMEAreceived()) {
+          GPS.parse(GPS.lastNMEA());
+        }
+        
+        if (GPS.fix) {
+          Serial.println(GPS.latitudeDegrees);
+          Serial.println(GPS.longitudeDegrees);
+          //data = GPS.latitude_fixed * (GPS.lat == 'N' ? 1 : -1) + 90 * 1E7;
```
Comment (lthiery) [inline longfi-us915/longfi-us915.ino]: What's this about? It look if we throw away the logic from these comments that we will have trouble running in Southern longitudes? It looks like the latitude logic needs help too.

**Review by lthiery — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -52,12 +52,13 @@ Select Tools -> Upload Method -> STM32CubeProgrammer(SWD)
 
 #### DFU
 Arduino IDE:  
-Select Tools -> Upload Method -> DFU  
+Select Tools -> Upload Method -> DFU
+Select Tools -> Serial Interface -> USB + HW Serial
```
Comment (lthiery) [inline README.md]: This is probably only for the Catena platform?

File: longfi-us915/longfi-us915.ino (hunk 1 above)
Comment (addy10) [reply longfi-us915/longfi-us915.ino]: The GPSSerial I think it is, the CFG_sx1276_radio can be remove

File: GPS/GPS.ino [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+void os_getDevKey (u1_t* buf) {  memcpy_P(buf, APPKEY, 16);}
+
+static uint8_t mydata[] = "Hello, world!";
+static osjob_t sendjob;
+
+// Schedule TX every this many seconds (might become longer due to duty
+// cycle limitations).
+const unsigned TX_INTERVAL = 60;
+
+// Pin mapping
```
Comment (lthiery) [inline GPS/GPS.ino:66]: Let's get rid of all the pin-mapping for boards we have not tested

File: GPS/GPS.ino [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+// This is the "App Key" in Helium. It is big-endian (msb).
+static const u1_t PROGMEM APPKEY[16] = { FILL_ME_IN };
+void os_getDevKey (u1_t* buf) {  memcpy_P(buf, APPKEY, 16);}
+
+static uint8_t mydata[] = "Hello, world!";
+static osjob_t sendjob;
+
+// Schedule TX every this many seconds (might become longer due to duty
+// cycle limitations).
+const unsigned TX_INTERVAL = 60;
```
Comment (lthiery) [inline GPS/GPS.ino]: Let's reduce this to 10 seconds to make things a little more interesting.

File: GPS/GPS.ino [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+              for (size_t i=0; i<sizeof(nwkKey); ++i) {
+                      if (i != 0)
+                              Serial.print("-");
+                      Serial.print(nwkKey[i], HEX);
+              }
+              Serial.println("");
+            }
+            // Disable link check validation (automatically enabled
+            // during join, but because slow data rates change max TX
+        // size, we don't use it in this example.
```
Comment (lthiery) [inline GPS/GPS.ino]: alignment

File: GPS/GPS.ino [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+          payload[idx++] = data >> 8;
+          payload[idx++] = data;
+          payload[idx++] = 0;
+          payload[idx++] = 0;
+        } else {
+          for (idx=0; idx<12; idx++) {
+            payload[idx] = 0;
+          }
+        }
+        //LMIC_setTxData2(1, mydata, sizeof(mydata)-1, 0);
```
Comment (lthiery) [inline GPS/GPS.ino]: dead code

**Review by lthiery — CHANGES_REQUESTED** [review]

File: GPS/GPS.ino [hunk 8]
```diff
@@ -0,0 +1,349 @@
...
+    .dio = {3, 6, LMIC_UNUSED_PIN},
+    .rxtx_rx_active = 0,
+    .rssi_cal = 8,              // LBT cal for the Adafruit Feather M0 LoRa, in dB
+    .spi_freq = 8000000,
+};
+
+#elif defined(MCCI_CATENA_4610) 
+#include "arduino_lmic_hal_boards.h"
+const lmic_pinmap lmic_pins = *Arduino_LMIC::GetPinmap_Catena4610();
+#elif defined(ARDUINO_DISCO_L072CZ_LRWAN1)
```
Comment (lthiery) [inline GPS/GPS.ino:86]: remove this one too and i think we're good to go

**Pull request merged**